server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/MailboxPathSorterTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.james.jmap.utils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.james.jmap.model.mailbox.Mailbox;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public class MailboxPathSorterTest {
+
+    @Test
+    public void sortFromRootToLeafShouldReturnOrderedMailbox() {
+        Mailbox inbox = Mailbox.builder().name("INBOX").id("INBOX").build();
+        Mailbox a = Mailbox.builder().name("A").id("A").parentId("INBOX").build();
+        Mailbox b = Mailbox.builder().name("B").id("B").parentId("INBOX").build();
+        Mailbox c = Mailbox.builder().name("C").id("C").parentId("B").build();
+        Mailbox d = Mailbox.builder().name("D").id("D").parentId("A").build();
+        Mailbox e = Mailbox.builder().name("E").id("E").parentId("C").build();
+
+        MailboxPathSorter sut = new MailboxPathSorter();
+        ImmutableList<Mailbox> input = ImmutableList.of(b, c, d, a, inbox, e);
+        List<String> result = sut.sortFromRootToLeaf(input)
```

Comment:
what is the action and what is the property extraction ? why don't you separate properties extraction from action ?